### PR TITLE
Domain Details Screen as WebView

### LINKED ISSUE #19071: 🤖 Domain Details Screen as WebView
A webview that displays the domain details.

## Screenshot Reference

Image

### PR DESCRIPTION
**Pull request by antonis**
Fixes #19071

## Desctiption
This PR redirects to a webview when the user taps on a domain in the domain management screen. The implementation reuses the `WPWebViewActivity` with a simpified UI.

https://github.com/wordpress-mobile/WordPress-Android/assets/304044/4c5ce1c0-6614-4e96-86e2-9f2d9c80d38c

## To test:
1. Enable the Domain Management feature flag (in debug settings)
2. Open the My domains screen: Me -> Domains -> My Domains
3. Tap on a domain
4. Verify that the `domain_management_my_domains_screen_domain_tapped` event is emmited
5. Verify that the domain details open in a webview

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

6. What automated tests I added (or what prevented me from doing so)
Added testcases for the new actions in the `DomainManagementViewModelTest`

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

File: WordPress/src/test/java/org/wordpress/android/ui/domains/DomainManagementViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    fun `WHEN a domain is tapped THEN track DOMAIN_MANAGEMENT_MY_DOMAINS_SCREEN_DOMAIN_TAPPED event`() {
+        viewModel.onDomainTapped(testDomain)
+        verify(analyticsTracker).track(DOMAIN_MANAGEMENT_MY_DOMAINS_SCREEN_DOMAIN_TAPPED)
+    }
+
+    @Test
+    fun `WHEN a domain is tapped THEN send DomainTapped action event`() = testWithActionEvents { events ->
+        viewModel.onDomainTapped(testDomain)
+        advanceUntilIdle()
+        assertThat(events.last()).isEqualTo(DomainManagementViewModel.ActionEvent.DomainTapped(testDomain))
```
Comment (mkevins) [inline WordPress/src/test/java/org/wordpress/android/ui/domains/DomainManagementViewModelTest.kt]: NP: `DomainManagementViewModel` qualifier can be omitted here (and on line 57 as well).

File: WordPress/src/test/java/org/wordpress/android/ui/domains/DomainManagementViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    private fun testWithActionEvents(block: suspend TestScope.(events: List<ActionEvent>) -> Unit) =
+        test {
+            val actionEvents = mutableListOf<DomainManagementViewModel.ActionEvent>()
+            val job = launch { viewModel.actionEvents.toList(actionEvents) }
```
Comment (mkevins) [inline WordPress/src/test/java/org/wordpress/android/ui/domains/DomainManagementViewModelTest.kt:61]: I noticed that this same pattern is used in the purchase screen tests, and I've observed some failures with those ([on a PR which didn't have changes there](https://github.com/wordpress-mobile/WordPress-Android/pull/19431)). I'm not sure if this is the same reason the tests are failing on this PR 🤔 .

Do we need to pass the `testDispatcher` to `launch` here? Afaiu, the `block` (i.e. the unit test's body) will suspend until the job is cancelled, but I'm wondering whether `advanceUntilIdle()` will suspend long enough to allow the events list to become populated before the assertion? 🤔

**Review by mkevins — COMMENTED** [review]
I have tested this, and it is working well for the most part (though there are some web-side changes needed to address some layout issues).

I also left a couple of comments about the tests.

For the web view changes, I noticed a few things that need to be addressed (probably in Calypso, and not in this PR):

For one of my domains, I was redirected to the "all domains" list, which also contains an entry point for adding a domain in the web view:



It's possible that this is because the domain is in an error state, but I wonder whether we should do anything when the webview tries to access URLs that we deem to be "invalid" for this flow? 🤔 

Another issue (which you may have already mentioned) is the padding at the top. I believe this has been addressed in the checkout page, but not for the details page, so we'll probably need a similar fix there.

And finally, there is the issue of the site creation flow entry point (web version), which I think should be omitted:



I will follow up about this one in the thread (pcdRpT-3PI-p2).

File: WordPress/src/test/java/org/wordpress/android/ui/domains/DomainManagementViewModelTest.kt (hunk 1 above)
Comment (antonis) [reply WordPress/src/test/java/org/wordpress/android/ui/domains/DomainManagementViewModelTest.kt]: Good catch @mkevins 👍 
Fixed with 78983567f5a9cea171f08d799a91f54c83c79b0d

File: WordPress/src/test/java/org/wordpress/android/ui/domains/DomainManagementViewModelTest.kt (hunk 2 above)
Comment (mkevins) [reply WordPress/src/test/java/org/wordpress/android/ui/domains/DomainManagementViewModelTest.kt:61]: It seems this was not the reason after all. The missing mock (also missing on my PR 😅 ) were the culprit, but the real error was masked by the unmatched `when` exception.

Thanks for resolving this Antonis!

**Review by mkevins — APPROVED** [review]
Thanks for updating this. This is working well, and the code changes look good. Since the other mentioned issues are related with the back-end, they are non-blocking here. Nice work!

**Comment by antonis** [comment]
Thank you for reviewing and testing this PR @mkevins 🙇 
I'll merge this PR to unblock any followup PRs and avoid conflicts.

> It's possible that this is because the domain is in an error state, but I wonder whether we should do anything when the webview tries to access URLs that we deem to be "invalid" for this flow? 🤔

I'll investigate this case further and iterate on a separate PR if this makes sense.

**Pull request merged**